service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformLogProcessingService.java

```diff
@@ -0,0 +1,168 @@
+package com.solace.maas.ep.event.management.agent.plugin.terraform.manager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.CommandRequest;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.CommandResult;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.JobStatus;
+import com.solace.maas.ep.event.management.agent.plugin.terraform.configuration.TerraformProperties;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class TerraformLogProcessingService {
+    public static final String KEY_LOG_LEVEL = "@level";
+    public static final String VALUE_LOG_LEVEL_ERROR = "error";
+    public static final String KEY_TYPE = "type";
+    public static final String VALUE_TYPE_APPLY_COMPLETE = "apply_complete";
+    public static final String VALUE_TYPE_APPLY_ERRORED = "apply_errored";
+    public static final String KEY_MESSAGE = "@message";
+    public static final String KEY_DIAGNOSTIC_DETAIL = "diagnosticDetail";
+    public static final String KEY_DIAGNOSTIC = "diagnostic";
+    private final TerraformProperties terraformProperties;
+    private final ObjectMapper objectMapper;
+
+    public TerraformLogProcessingService(ObjectMapper objectMapper, TerraformProperties terraformProperties) {
+        this.objectMapper = objectMapper;
+        this.terraformProperties = terraformProperties;
+    }
+
+    public void saveLogToFile(CommandRequest request, List<String> logs) throws IOException {
+        Path logPath = Paths.get(terraformProperties.getWorkingDirectoryRoot()
+                + File.separator
+                + request.getContext()
+                + "-"
+                + request.getServiceId()
+                + File.separator
+                + "logs"
+        );
+
+        if (Files.notExists(logPath)) {
+            Files.createDirectories(logPath);
+        }
+
+        Path out = Files.createTempFile(logPath, System.currentTimeMillis() + "-"
+                + request.getCorrelationId() + "-job", ".log");
+
+        Files.write(out, logs, Charset.defaultCharset());
+    }
+
+    public CommandResult buildTfCommandResult(List<String> jsonLogs) {
+        if (CollectionUtils.isEmpty(jsonLogs)) {
+            throw new IllegalArgumentException("Cannot extract information empty logs");
```

Comment:
In case there are no logs, would we return a response back to EP ? or does this exception prevents any response being built and sent back to EP?

Reply:
Yes, the error handling needs some re-work. I'll fix it up.